adodb.inc.php

```diff
@@ -973,60 +974,91 @@ abstract protected function _connect($argHostname, $argUsername, $argPassword, $
 	/**
 	 * Connect to database.
 	 *
-	 * @param string $argHostname     Host to connect to
-	 * @param string $argUsername     Userid to login
-	 * @param string $argPassword     Associated password
+	 * @param string $argHostname Host to connect to
+	 * @param string $argUsername Userid to login
+	 * @param string $argPassword Associated password
 	 * @param string $argDatabaseName Database name
-	 * @param bool   $forceNew        Force new connection
-	 *
+	 * @param bool $forceNew Force new connection
+	 * @param null|int $port (Optional) The port to use for the connection.
+     * @param null|string $socket (Optional) The socket to use for the connection.
 	 * @return bool
 	 */
-	function Connect($argHostname = "", $argUsername = "", $argPassword = "", $argDatabaseName = "", $forceNew = false) {
+	function Connect($argHostname = "", $argUsername = "", $argPassword = "", $argDatabaseName = "", $forceNew = false, $port = null, $socket = null) {
+		
 		if ($argHostname != "") {
 			$this->host = $argHostname;
 		}
+		
 		// Overwrites $this->host and $this->port if a port is specified.
 		$this->parseHostNameAndPort();
 
+		// User defined port, PHP7 compat: port must be int.
+		if (!is_null($port)) {
+
+			// Use default port if cast yields zero.
+			$port = (int)$port != 0 ? (int)$port : 3306;
+		
+		} else {
+
+			// Use default port if cast yields zero.
+			$port = (int)$this->port != 0 ? (int)$this->port : 3306;
+		}
```

Comment:
This does not belong in the parent method - port 3306 is a MySQL-specific thing.

Reply:
Reverted changes not related to fixing the inappropriate setting of the socket, adding a check to make sure something isn't already set for the clientFlags before setting a default, and updating the default to verify the server cert for better default security for SSL. Created a second pull request for updates to setConnectionParameter 
 from our discussion https://github.com/ADOdb/ADOdb/pull/1044